### PR TITLE
Rel 4.9.0.4

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Comment by miczyg1** [comment]
@artur-rs could You please fill validation statistics and what has been changed/improved in terms of tests?

**Comment by miczyg1** [comment]
@pietrushnic do You have anything to add?

File: _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+   **ROCA TPM vulnerability** for Infineon SLB9665 chips present on
+   **TPM1a modules**. The engineers verified that the **TPMs are vulnerable**.
+   For details and mitigation refer to the [documentation](https://github.com/pcengines/apu2-documentation/blob/master/docs/research/ROCA.md).
+   The document describes how to **upgrade TPM firmware** and
+   **get rid of the vulnerability**.
+7. Added a [list of supported and verified mPCIe modules](https://github.com/pcengines/apu2-documentation/blob/master/docs/mpcie_modules.md).
+   The list also describes **known issues** with modules and **workarounds** to get the **modules working** properly. Please check the list when
+   encountered any issues with mPCIe modules. Contribution is welcome. Also be
+   sure to check whether the slot is compatible with the module. In order to do so, look at the interfaces connected to each mPCIe slot [here](https://github.com/pcengines/apu2-documentation/blob/master/docs/APU_mPCIe_capabilities.md) and the interfaces required by the module in its
+   product specification etc.
```
Comment (pietrushnic) [inline _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md]: @miczyg1 80 char wrapping?

File: _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+```
+git diff --stat 7a732b4 ':(exclude)gitlab-ci.yml' ':(exclude)CHANGELOG.md'
+```
+
+`72 files changed, 2240 insertions(+), 138 deletions(-)`
+
+![Process of mainlining](https://cloud.3mdeb.com/index.php/s/c5rmF8zXpkzYcHx/preview)
+
+The chart represents the total line added and deleted on the PC Engines
+corebootÂ fork against the rebase point for given release. Check the statistics
```
Comment (pietrushnic) [inline _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md]:
```suggestion
coreboot fork against the rebase point for a given release. Check the statistics
```

File: _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md (hunk 1 above)
Comment (miczyg1) [reply _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md]: fixed

File: _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md (hunk 2 above)
Comment (miczyg1) [reply _posts/2019-04-09-PC-Engines-Firmware-v4-9-0-4.md]: fixed

**Pull request merged**